static/keyboard.js

```diff
@@ -211,46 +236,53 @@ function Keyboard() {
   // search //
   ////////////
 
-  const [shownIndexes, setShownIndexes] = useState([]);
+  /** list of search results, blank if none */
+  const [searchResults, setSearchResults] = useState([]);
   const [query, setQuery] = useState("");
+  /** list of targets to search on */
   const targets = useRef(null);
-  const allIndexes = useRef([]);
   const keys = ["name", "description", "overloads"];
 
   useEffect(() => {
-    const indexes = [];
     targets.current = Object.entries(codepage_descriptions).flatMap(
       ([index, elts]) => {
-        indexes.push(index.toString());
         return elts.map((elt) => {
-          const result = { index };
+          const result = { index, token: elt.token };
           keys.forEach((key) => (result[key] = fuzzysort.prepare(elt[key])));
           return result;
         });
       }
     );
-    allIndexes.current = [...new Set(indexes)];
   }, []);
 
   useEffect(() => {
-    if (query) {
-      const results = fuzzysort.go(query, targets.current, {
+    setSearchResults(
+      fuzzysort.go(query, targets.current, {
+        all: true,
         keys: ["name", "description", "overloads"],
-      });
-      setShownIndexes([...new Set(results.map((result) => result.obj.index))]);
-    } else {
-      setShownIndexes(allIndexes.current);
-    }
+        threshold: -10000,
+      })
+    );
   }, [query]);
 
   const renderChildren = () => {
-    return [...shownIndexes].map((i) => {
+    const keys = [
+      ...searchResults
+        .reduce((map, result) => {
+          if (!map.has(result.obj.index)) map.set(result.obj.index, []);
+          map.get(result.obj.index).push(result);
+          return map;
+        }, new Map())
+        .entries(),
+    ];
+    return keys.map(([i, results]) => {
       const chr = codepage[i];
       return html`<${Tooltip}
         key=${i}
-        shown=${shownIndexes.includes(i)}
+        shown=${true}
```

Comment:
oh lol i should remove this

Reply:
often when I'm too lazy to open up vscode I just slam suggestions at myself
```suggestion
        shown=${shownIndexes.includes(i)}
```

Reply:
no i mean to just remove the argument lol